src/language/dafnyInstallation.ts

```diff
@@ -24,32 +28,65 @@ function ifNullOrEmpty(a: string | null, b: string): string {
   return a === null || a === '' ? b : a;
 }
 
-function getConfiguredVersion(): string {
-  const version = Configuration.get<string>(ConfigurationConstants.PreferredVersion);
-  return version === LanguageServerConstants.Latest
-    ? LanguageServerConstants.LatestVersion
-    : version;
+async function getConfiguredVersion(): Promise<string> {
+  const [ _, version ] = await getConfiguredTagAndVersion();
+  return version;
+}
+
+let getConfiguredTagAndVersionCache: [string, string];
+async function getConfiguredTagAndVersion(): Promise<[string, string]> {
+  if(getConfiguredTagAndVersionCache === undefined) {
+    const result = await getConfiguredTagAndVersionUncached();
+    if(getConfiguredTagAndVersionCache === undefined) {
+      getConfiguredTagAndVersionCache = result;
+    }
+  }
+  return getConfiguredTagAndVersionCache;
+}
+
+async function getConfiguredTagAndVersionUncached(): Promise<[string, string]> {
```

Comment:
```suggestion
// Returns [git tagName, version displays name]
async function getConfiguredTagAndVersionUncached(): Promise<[string, string]> {
```

Reply:
How about putting that information in the method name, then it doesn't go out of date so easily? 
Do you want to rename `getConfiguredTagAndVersionUncached` to `getConfiguredGitTagAndVersionUncached` ?

Reply:
Ok, yes, the fact that the tag is a git tag was important to me. Funny enough, when I was looking at the signature, I was trying to understand what the return type was without looking at the function's name, so I guessed it based on the function's content.